### ISSUE TITLE
Genotyping with information of recepient 

### ISSUE THREAD

**Issue by flde** [single-cell-genetics/vireo#62]
Hi all, many thanks for the userfriendly implementation of vireo! 

I hope you could give me some advice on the following project. I have single cell RNAseq data from donor/recepient over four time points. The first two time points are recepient cells only the last two time points are mixed donor/recepient cells which I would like to distinguish. 

My strategy would be to run cellsnp-lite on the four BAM files first (=four time points). Next, I can extract the significant recepient SNPs from the first two time points with bcftools 

`bcftools view recepient.vcf.gz -R cellSNP.cells.vcf.gz -Oz -o sub.vcf.gz`

and hand that list to vireo -d parameter while the -c cell data is from the third and fourth time point (=mixed time points)

`vireo -c $CELL_DATA -d $RECEPIENT_GT_FILE -o $OUT_DIR -N $n_recepient`

Does that make sense or do I miss something? I highly appreciate you help! 

Best, 
Florian

**Comment by huangyh09** [comment]
Hi Florian,

Thanks for your questions and your experiment settings. It looks like you only have genotypes from the recipient but not the donor, so the use of `$RECEPIENT_GT_FILE` may be not the best option.

The option you mentioned by pooling all four samples and demultiplexed by Vireo and then using the recipient cells as QC matrix is actually a very good one.

Alternatively, you could do it on the two samples and check the genotype differences between the recipient, e.g., with this [donor matching tutorial](https://nbviewer.org/github/single-cell-genetics/vireo/blob/master/examples/donor_match.ipynb); I assume you genotyped with cellsnp-lite in a pseudo-bulk manner).

Yuanhua

**Comment by flde** [comment]
Hi @huangyh09, 

Many thanks for your quick response. I will try the pooling approach first so that I get familiar with the pipeline. 

Regarding the second approach, I did not configure cellsnp-lite as pseudo-bulk for the recipient samples. But that makes a lot of sense since it increase the coverage per gene, right? I don't fully understand the tutorial yet. Would you compare the genotype differences across recepients and use then only high confindent areas to deconvolut the recepient/donor samples respectivly? 

Best wishes, Florian

**Comment by flde** [comment]
Hi @huangyh09, 

I pooled the BAM per patient over all four samples (baseline, day0, day14, day100) and run cellsnp-light + vireon as suggested. In the first two samples we only have recepient cells while the later samples are a mixture of recepient and donor. The results look really good for patient 766, 768, 784, 785. For patient 764 we only have recepient cells which makes the analysis obsolet. 

However, for patient 763 and 783 the results do not look as good. We know that the cell transplant for patient 763 failed so we do not expect many donor cells at day14.

I went through the tutorials looking for options to increase the sensitivity of the approach. Do you have any recomendation what I could try to recover those samples? I have a maximum of 48 threads on the cluster in case its computational heavy. 

The performance of the tool on the good samples (many cells / good recepient to donor balance) is really nice however. Many thanks!

![download](https://user-images.githubusercontent.com/20926246/178418480-b614d388-4443-4724-acea-3a49e7ea5e2a.png)


![download](https://user-images.githubusercontent.com/20926246/178418480-b614d388-4443-4724-acea-3a49e7ea5e2a.png)

**Comment by flde** [comment]
Hi @huangyh09, 

I talked with my colaborators and they told me that for some patients the donor/recepient pairs are close relatives. I think 783 is a parent-child combination and 784 are siblings. In case of 763 the transplant probably failed so its highly likely only one genotype. 

I repeated snp-lite with an allel frequency of 0.05 instead of 0.1. In addition I set -M 500 since it only takes 20 min. For plotting I filtered the cells for >1500 UMI | < 15 MT% | removed doublets based on HTO. 

I am actually stunned about how well that worked! I am even tempted to use the results without doublet detection but not sure if they are robust e.g. there is some miss-classification on the baseline with 783. 

Many thanks for your dedication and input, Yuanhua. I think I can close the issue now but if you have another advice its highly appreciated. All the best!

![download](https://user-images.githubusercontent.com/20926246/180195601-e185b50d-20b6-4ef5-9e81-24678f91af6e.png)
![download](https://user-images.githubusercontent.com/20926246/180196825-a396884a-15fe-4bf6-83b1-095544f85466.png)
